service/frontend/workflow_handler.go

```diff
@@ -4743,6 +4780,31 @@ func (wh *WorkflowHandler) cancelOutstandingPoll(ctx context.Context, namespaceI
 	return true
 }
 
+func (wh *WorkflowHandler) registerOutstandingPollContext(
+	ctx context.Context,
+	pollerID string,
+	namespaceID string,
+) context.Context {
+
+	if pollerID != "" {
+		childCtx, cancel := context.WithCancel(ctx)
+		nsPollers, _ := wh.outstandingPollers.GetOrSet(namespaceID, collection.NewSyncMap[string, context.CancelFunc]())
+		nsPollers.Set(pollerID, cancel)
+		return childCtx
+	}
+	return ctx
+}
+
+func (wh *WorkflowHandler) unregisterOutstandingPollContext(
+	pollerID string,
+	namespaceID string,
+) {
+	nsPollers, ok := wh.outstandingPollers.Get(namespaceID)
+	if ok {
+		nsPollers.Delete(pollerID)
```

Comment:
shouldn't you always call `cancel()` after you create a context with `WithCancel`? this causes a leak otherwise?

Reply:
yes. called cancel()